.github/workflows/v1.yml

```diff
@@ -0,0 +1,55 @@
+name: Run Tests
+
+on:
+  pull_request:
+    branches:
+      v1
+
+env:
+  GO111MODULE: on
+  GOPROXY: https://proxy.golang.org
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        go: [1.11, 1.12, 1.13]
+    name: ${{ matrix.os }} @ Go ${{ matrix.go }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Set up Go ${{ matrix.go }}
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Set GOPATH and PATH
+        run: |
+          echo "##[set-env name=GOPATH;]$(dirname $GITHUB_WORKSPACE)"
+          echo "##[add-path]$(dirname $GITHUB_WORKSPACE)/bin"
+        shell: bash
+
+      - name: Checkout Code
+        uses: actions/checkout@v1
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Install Dependencies
+        run: |
+          go get github.com/urfave/gfmrun/cmd/gfmrun
+          go get golang.org/x/tools/cmd/goimports
+          go mod tidy
```

Comment:
I'm assuming the reason to run `go mod tidy` is because installing the CLI tools inside the working directory messes up the `go.mod` file. The downside to this pattern is we're making unnecessary modifications to `go.mod`, and after this point, we can't verify that the `go.mod` was actually tidied before the PR, which is probably something we want to do.

There's a few different ways to handle this:
- Install binaries instead of building from source. I don't think `goimports` distributes a binary, but we can probably add one to `gfmrun` at some point. I prefer to do this when possible.
- Install with `GO111MODULE=off`. This avoids writing the `go.mod` file, but has the downside of ignoring a dependency's `go.mod` if it exists. I typically do this if they don't have a binary, since it's pretty unusual for Go CLIs to break `go get` installations without providing an alternative installation method. 
- Create a temporary directory, do a `go mod init tmp` there, then do `go get` on CLI dependencies. This is a few extra lines to do, but it does respect the `go.mod` for transitive dependencies.

Reply:
You are right in guessing the reason about using `go mod tidy`. However we do not need to worry about modifying the `go.mod` accidentally. We use `git diff --exit-code` at the end of our CI process which ensures that there are no files modified.

About `goimports`it is carried forward from a legacy build process. I think if everyone thinks that it's useless, I'd be more than happy to remove it. I although am in favour of updating the `gfmrun` project to build and release a binary artifact and then use that artifact in our build process.

Once that is done, I think even `go mod tidy` will become redundant.

Reply:
I actually use `goimports` in all my projects and usually enforce it using `golangci-lint`, although you could enforce it by just running `goimports` and checking the diff. So I'm 👍 if we want to add it back into the pipeline.

A quick grep of the repo makes it look like we aren't actually using `goimports` anywhere, so we can either remove it or table it for now and deal with that separately if we want to keep the scope of this change set smaller.

Reply:
`goimports` has been removed and now we are using the pre-compiled binary for `gfmrun`